databricks_cli/unity_catalog/cli.py

```diff
@@ -0,0 +1,65 @@
+# Databricks CLI
+# Copyright 2021 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+
+from databricks_cli.configure.config import profile_option, debug_option
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS
+from databricks_cli.version import print_version_callback, version
+
+from databricks_cli.unity_catalog.metastore_cli import register_metastore_commands
+from databricks_cli.unity_catalog.catalog_cli import register_catalog_commands
+from databricks_cli.unity_catalog.schema_cli import register_schema_commands
+from databricks_cli.unity_catalog.table_cli import register_table_commands
+from databricks_cli.unity_catalog.ext_loc_cli import register_ext_loc_commands
+from databricks_cli.unity_catalog.cred_cli import register_cred_commands
+from databricks_cli.unity_catalog.delta_sharing_cli import register_delta_sharing_commands
+from databricks_cli.unity_catalog.perms_cli import register_perms_commands
+
+
+@click.group(context_settings=CONTEXT_SETTINGS,
+             help='Utility to interact with Databricks unity-catalog.\n\n' +
+             '**********************************************************************\n' +
+             'WARNING: these commands are EXPERIMENTAL and not officially supported.\n' +
+             '**********************************************************************',
+             hidden=True)
+@click.option('--version', '-v', is_flag=True, callback=print_version_callback,
+              expose_value=False, is_eager=True, help=version)
+@debug_option
+@profile_option
+@eat_exceptions
+def unity_catalog_group():  # pragma: no cover
+    """
+    Utility to interact with Databricks unity-catalog.
+    """
+    pass
+
+
+register_metastore_commands(unity_catalog_group)
```

Comment:
I can see that being more consistent with the databricks-cli command hierarchy. However, it would double set of root commands (adding ~12 to the current 15) by adding: 
- metastore
- catalog
- schema
- table
- external-location
- storage-credential
- share
- provider
- recipient
- table-lineages
- column-lineages
- functions (TBA?)

I'd also be concerned about possible customer confusion, since the above root commands only apply to UC objects (i.e., a customer might think `table` commands apply to Hive Metastore).
So I'd suggest gathering more input on this proposal.

Reply:
don't we want to eventually enable UC for all of the customers?

Reply:
All subgroups still pertain to Unity Catalog so I think it's logical to keep them under "unity-catalog". A couple of groups are even ambiguous if we were to hoist them to the top level, e.g. "share".